trino/dbapi.py

```diff
@@ -199,17 +201,21 @@ def _create_request(self):
             client_tags=self.client_tags
         )
 
-    def cursor(self, experimental_python_types=False):
+    def cursor(self, experimental_python_types: bool = None):
         """Return a new :py:class:`Cursor` object using the connection."""
         if self.isolation_level != IsolationLevel.AUTOCOMMIT:
             if self.transaction is None:
                 self.start_transaction()
-            request = self.transaction._request
-        elif self.transaction is not None:
-            request = self.transaction._request
+        if self.transaction is not None:
+            request = self.transaction.request
         else:
             request = self._create_request()
-        return Cursor(self, request, experimental_python_types)
+        return Cursor(
+            self,
+            request,
+            # if experimental_python_types is not explicitly set in Cursor, take from Connection
```

Comment:
Could you add a test case, experimental_python_types=True in connection and experimental_python_types=False in cursor?